ossia-score: init at 3.2.3

jcelerier opened the pull request:
## Description of changes

Continuing the PR https://github.com/NixOS/nixpkgs/pull/174802 which looks abandoned

@magnetophon @minijackson @ctem @luzpaz 


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

minijackson committed:
ossia-score: init at 3.2.3

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
Co-authored-by: Jean-Michaël Celerier <jeanmichael.celerier@gmail.com>
Co-authored-by: Rémi NICOLE <minijackson@users.noreply.github.com>

jcelerier committed:
maintainers: add jcelerier

After discussion on nixpkgs-audio matrix channel (@minijackson)

jcelerier commented:
@vlnk you think you would have some time to try building this PR ? I don't have a Nix environment easily accessible :|

jcelerier commented:
rebased to a closer tip

minijackson commented:
Yeah sorry, haven't had the time to update my PR, so it's good if someone else picks it back up. Thanks a lot!

Some notes:

this PR depends on #277145 as it uses `lib.maintainers.jcelerier`, so this doesn't build currently.

For the commit conventions, an package update should look like `$package: $oldversion -> $newversion`, but since it both adds the package and updates it in the same PR, I think it's best to squash both commits, with description: `ossia-score: init at 3.2.0`.

jcelerier commented:
okay, squashed and force-pushed ! do you think you could try to build to check if there's any issue ?

minijackson commented:
Trying without `jcelerier` leads to two issues:

I get `sha256-xmPHEACIay98p5YgZyAi7NQm9vaNZwOtyZsyZWyQsJA=` as src hash.

CMake fails with:

```
CMake Error at /nix/store/q1nssraba326p2kp6627hldd2bhg254c-cmake-3.29.2/share/cmake-3.29/Modules/ExternalProject.cmake:2945 (message):
  error: could not find git for clone of libcoap-populate
```

If ossia-score needs libcoap-populate, it probably needs to be also packaged for nixpkgs, I haven't found it.

jcelerier commented:
also, if there's any way I can easily add Nix builds to my github actions matrix, that would be great.

- here are the per-distro build scripts: https://github.com/ossia/score/tree/master/ci 
- the GH action script is pretty simple: https://github.com/ossia/score/blob/master/.github/workflows/builds.yaml#L106

minijackson commented:
That would be pretty nice! I think you would have to have a `.nix` file very similar to the one you're currently contributing, with two main differences:

- use `src = ./.;` (or `src = ../.;` if from a subdirectory) instead of using `fetchFromGitHub ...`
- you'll have to import nixpkgs since it would be a separate repo

You could use `https://github.com/cachix/install-nix-action` to install Nix for your GitHub action.

minijackson commented:
Note if you want to test this PR yourself @jcelerier, [installing Nix](https://nixos.org/download/) should be straightforward, and works on any Linux distribution, and doesn't have impact, as everything usually stays inside `/nix`

SuperSandro2000 commented on pkgs/applications/audio/ossia-score/default.nix:
```diff
@@ -0,0 +1,125 @@
+
+  meta = with lib; {
+    homepage = "https://ossia.io/score/about.html";
+    description = "A sequencer for audio-visual artists, designed to enable the creation of interactive shows, museum installations, intermedia digital artworks, interactive music and more in an intuitive user interface";
```
```suggestion
    description = "A sequencer for audio-visual artists, designed to enable the creation of interactive shows, museum installations, intermedia digital artworks, interactive music and more";
```
that's quite a bit long, lets try to shorten that without losing important infos

SuperSandro2000 commented on pkgs/applications/audio/ossia-score/default.nix:
```diff
@@ -0,0 +1,125 @@
+  installPhase = ''
+    runHook preInstall
+
+    cmake -DCMAKE_INSTALL_DO_STRIP=1 -DCOMPONENT=OssiaScore -P cmake_install.cmake
```
We cannot use cmakeFlags here?

jcelerier replied:
hmm these aren't the flags used for the build step, but for the install step

jcelerier replied:
I don't think we can, these aren't the flags passed at CMake configure time (what cmakeFlags seems to be for), but the flags passed post-build, for installation, which are very much unrelated. e.g. this tells cmake to run the script "cmake_install.cmake" with specific parameters (specifically, we only want to install a specific part of the ossia score build tree, not the default which installs a ton of things that are not meaningful for a deployment in a Linux distro)

jcelerier commented:
I'm trying to build it locally but it's failing during a CMake FetchContent operation with :

```
CMake Error at /nix/store/q1nssraba326p2kp6627hldd2bhg254c-cmake-3.29.2/share/cmake-3.29/Modules/FetchContent.cmake:1714 (message):
  Build step for libcoap failed: 1
Call Stack (most recent call first):
  /nix/store/q1nssraba326p2kp6627hldd2bhg254c-cmake-3.29.2/share/cmake-3.29/Modules/FetchContent.cmake:1854:EVAL:2 (__FetchContent_directPopulate)
  /nix/store/q1nssraba326p2kp6627hldd2bhg254c-cmake-3.29.2/share/cmake-3.29/Modules/FetchContent.cmake:1854 (cmake_language)
  /nix/store/q1nssraba326p2kp6627hldd2bhg254c-cmake-3.29.2/share/cmake-3.29/Modules/FetchContent.cmake:2081 (FetchContent_Populate)
  3rdparty/libossia/cmake/deps/coap.cmake:37 (FetchContent_MakeAvailable)
  3rdparty/libossia/cmake/OssiaDeps.cmake:107 (include)
  3rdparty/libossia/CMakeLists.txt:36 (include)
```

jcelerier commented:
Managed to build & run it locally with https://github.com/ossia/score/blob/master/ci/nix.build.nix
Updated the PR to use more libraries from the system instead of the builtin ones

magnetophon commented:
I am unable to build this:



```

source> remote: Compressing objects: 100% (241/241), done.
source> remote: Total 718 (delta 566), reused 548 (delta 470), pack-reused 0
source> Receiving objects: 100% (718/718), 272.71 KiB | 4.20 MiB/s, done.
source> Resolving deltas: 100% (566/566), completed with 370 local objects.
source> From https://github.com/jcelerier/vst3_public_sdk
source>  * branch            a0e728e942103b01e3c894b9f7bb1c64f0686f24 -> FETCH_HEAD
source> Submodule path '3rdparty/vst3/public.sdk': checked out 'a0e728e942103b01e3c894b9f7bb1c64f0686f24'
source> Submodule path '3rdparty/zipdownloader': checked out '57d4394ec4fcaa3995670abd0aa94e36b69dcc60'
source> remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
source> remote: Enumerating objects: 5, done.
source> remote: Counting objects: 100% (5/5), done.
source> remote: Compressing objects: 100% (3/3), done.
source> remote: Total 3 (delta 2), reused 1 (delta 0), pack-reused 0
source> Unpacking objects: 100% (3/3), 395 bytes | 395.00 KiB/s, done.
source> From https://github.com/Velron/doxygen-bootstrapped
source>  * branch            4bfd385c1a5c1ae163f445ed1c84062d25b73147 -> FETCH_HEAD
source> Submodule path 'docs/Doxygen/doxygen-bootstrapped': checked out '4bfd385c1a5c1ae163f445ed1c84062d25b73147'
source> removing `.git'...
error: build of '/nix/store/lrb7zh3gm49rjv9n5yzd0f9n4h8maqzb-source.drv' on 'ssh://nixBuild@builder' failed: hash mismatch in fixed-output derivation '/nix/store/lrb7zh3gm49rjv9n5yzd0f9n4h8maqzb-source.drv':
         specified: sha256-j/4zAFDJi9LX8hs625TlcP/BJEZWFd1LHsAy/k/yXEs=
            got:    sha256-xmPHEACIay98p5YgZyAi7NQm9vaNZwOtyZsyZWyQsJA=
            
            ```

minijackson commented:
>  Updated the PR to use more libraries from the system instead of the builtin ones

thanks for doing that, it really helps us when a new security vulnerability comes to light.

Some things that come to mind:

- if you're still in the process of developing the PR, I think it's good to to mark it as draft
- since I started the PR, there's a new convention in nixpkgs on the location of package definition. Could you move `pkgs/applications/audio/ossia-score/default.nix` to `pkgs/by-name/os/ossia-score/package.nix`, and remove the modifications from `pkgs/top-level/all-packages.nix`? This would allow us to use [nixpkgs-merge-bot](https://github.com/NixOS/nixpkgs/blob/b924370a5ec14549285da9c36e4f2cbd2e722f62/CONTRIBUTING.md#how-to-merge-pull-requests)

jcelerier commented:
> > Updated the PR to use more libraries from the system instead of the builtin ones
> 
> thanks for doing that, it really helps us when a new security vulnerability comes to light.
> 
> Some things that come to mind:
> 
>     * if you're still in the process of developing the PR, I think it's good to to mark it as draft
> 
>     * since I started the PR, there's a new convention in nixpkgs on the location of package definition. Could you move `pkgs/applications/audio/ossia-score/default.nix` to `pkgs/by-name/os/ossia-score/package.nix`, and remove the modifications from `pkgs/top-level/all-packages.nix`? This would allow us to use [nixpkgs-merge-bot](https://github.com/NixOS/nixpkgs/blob/b924370a5ec14549285da9c36e4f2cbd2e722f62/CONTRIBUTING.md#how-to-merge-pull-requests)

done :) I don't think I have any others modifications to do. I'm trying to reproduce the build failure of @magnetophon but no luck so far.

jcelerier commented:
Hmm some chicks are failing due to moving to the by-name organization.. could you check @minijackson ? I have no clue what's wrong :/

jcelerier commented:
added some more deps as third parties (all the remaining ones are heavily-rewritten forks of existing libraries or not available in nix in a version that is compatible with ossia from what I can see)

jcelerier commented:
Note : on aarch64 ossia can build but only with clang (due to the use of https://github.com/kfrlib/kfr for dsp, which itself only supports clang, not gcc there)

minijackson commented:
I can now run it and and do some stuff with it, thanks! While testing it with [score-examples](https://github.com/ossia/score-examples), I found some bugs, (visual glitches, some texture wouldn't be created), but I'm not sure whether it comes from the packaging.

---

Result of `nixpkgs-review pr 315409` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ossia-score</li>
  </ul>
</details>

minijackson commented on pkgs/by-name/os/ossia-score/package.nix:
```diff
@@ -0,0 +1,137 @@
+    homepage = "https://ossia.io/score/about.html";
+    description = "A sequencer for audio-visual artists, designed to enable the creation of interactive shows, museum installations, intermedia digital artworks, interactive music and more";
+    # TODO: this should work for darwin too
+    platforms = platforms.linux;
```
Since you said aarch64-linux needs some extra care:

```suggestion
    platforms = "x86_64-linux";
```

minijackson commented on pkgs/by-name/os/ossia-score/package.nix:
```diff
@@ -0,0 +1,137 @@
+    # TODO: this should work for darwin too
+    platforms = platforms.linux;
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ minijackson jcelerier ];
```
alphabetically sorted:

```suggestion
    maintainers = with maintainers; [ jcelerier minijackson ];
```

minijackson commented on pkgs/by-name/os/ossia-score/package.nix:
```diff
@@ -0,0 +1,137 @@
+  src = fetchFromGitHub {
+    owner = "ossia";
+    repo = "score";
+    rev = "fd8efedb29be05ce3b7ea067fdfdcae6537ebb47";
```
This doesn't seem to be on the 3.2.0 tag: https://github.com/ossia/score/commit/ea4e943c6f7819824a24b4de5226b6c822debd06

If there are some upstream changed that we need, we can add them through patching, for example like this:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/waylandpp/default.nix#L28-L38

jcelerier replied:
any way to cherry-pick a specific commit instead of a patch file ? it's these two patches in the libossia submodule which are necessary :
https://github.com/ossia/libossia/commit/23f64e41473905d4b5c8c0267140dc60412253ee
https://github.com/ossia/libossia/commit/130e64f916c6805b7bb6bae5617f59051e968e6c

minijackson replied:
Yes, like shown in the link, you can do something like this:

```nix
  patches = [
    # A nice comment explaining why this is needed
    (fetchpatch {
      name = "fix-missing-libsamplerate.patch";
      url = "https://github.com/ossia/libossia/commit/23f64e41473905d4b5c8c0267140dc60412253ee.patch";
      hash = ""; # to be filled
    })

    # Another comment
    (fetchpatch {
      # ...
    })
  ];
```

You will need to add `fetchpatch` in the arguments at the top.

jcelerier replied:
oh ! i didn't know this github feature at all

jcelerier replied:
hmm.. the problem is : the patch applies to a submodule, not to the main repo. I tried setting a path with: 

    extraPrefix = "/3rdparty/libossia";

in the fetchpatch invocation but to no avail.

jcelerier replied:
okay, found another way

minijackson replied:
Can you try `extraPrefix = "3rdparty/libossia/";` instead? It seems to be what people are doing from a quick look inside nixpkgs

magnetophon commented:
My build failure seems to have been a glitch:

Result of `nixpkgs-review pr 315409` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ossia-score</li>
  </ul>
</details>

minijackson commented:
> My build failure seems to have been a glitch

@magnetophon I think the `src` was updated between your two tests

https://github.com/NixOS/nixpkgs/compare/1e94a77fe605485ad0b8040340da5c8a7303c533..3e6c79bf3498c893c00d38a88a0b3e6abe695a78

jcelerier commented on pkgs/by-name/os/ossia-score/package.nix:
```diff
@@ -130,7 +130,7 @@ stdenv.mkDerivation (finalAttrs: {
     description = "A sequencer for audio-visual artists, designed to enable the creation of interactive shows, museum installations, intermedia digital artworks, interactive music and more";
     # TODO: this should work for darwin too
-    platforms = platforms.linux;
+    platforms = "x86_64-linux";
```
@minijackson it seems that this change broke the build ?

minijackson replied:
Oh right sorry, it's supposed to be a list:

```nix
    platforms = [ "x86_64-linux" ];
```

jcelerier commented:
for the aarch64 build, any "standard" way to tell it to use clang for building instead of gcc ?

minijackson commented:
It might be possible to do something like this:

```nix
let
  env = if stdenv.hostPlatform.isAarch64 then clangStdenv else stdenv;
in
env.mkDerivation {
  ...
```

jcelerier commented:
>  ossia-score, ossia-score.passthru.tests on aarch64-darwin Queued — Waiting to run this check...

idk why it's queued, it should definitely not try running on Darwin at this point (likely this can work too but with some adaptations on the packages)

jcelerier commented:
after this, what remains ?

minijackson reviewed (approved):
Couldn't test it due to bad internet connection, but looks good! Thanks a lot!

h7x4 commented on pkgs/by-name/os/ossia-score/package.nix:
```diff
@@ -0,0 +1,144 @@
+, avahi
+, bluez
+, boost185
+# , faust
```
Is this supposed to be commented out? If so, could you leave a small comment about why it is here?

jcelerier replied:
hmm this was in the previous PR i sourced the package.nix from. This is an optional feature which apparently doesn't work yet in nix.

magnetophon reviewed (approved):
Result of `nixpkgs-review pr 315409` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ossia-score</li>
  </ul>
</details>

jcelerier commented:
up @minijackson

jcelerier commented:
otherwise I released 3.2.1 which removes the need for the patches that come with the package, but if this delays the merge for another couple weeks.. I'd rather make a subsequent patch on top of a working basis :)

Aleksanaa commented on pkgs/by-name/os/ossia-score/package.nix:
```diff
@@ -0,0 +1,142 @@
+
+  meta = with lib; {
+    homepage = "https://ossia.io/score/about.html";
+    description = "A sequencer for audio-visual artists, designed to enable the creation of interactive shows, museum installations, intermedia digital artworks, interactive music and more";
```
```suggestion
    description = "Sequencer for audio-visual artists";
    longDescription = ''
      ossia score is a sequencer for audio-visual artists, designed to enable
      the creation of interactive shows, museum installations, intermedia
      digital artworks, interactive music and more in an intuitive user interface.
    '';
```

Aleksanaa commented on pkgs/by-name/os/ossia-score/package.nix:
```diff
@@ -0,0 +1,142 @@
+  meta = with lib; {
+    homepage = "https://ossia.io/score/about.html";
+    description = "A sequencer for audio-visual artists, designed to enable the creation of interactive shows, museum installations, intermedia digital artworks, interactive music and more";
+    platforms = [ "x86_64-linux" ];
```
Not even `platforms.linux`? It supports raspberry pi though

jcelerier replied:
It builds on aarch64 but only with clang, not GCC (check the messages above in the thread)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/1016/157

minijackson commented:
So I can tell that it compiles and that it runs, but I'm having trouble making it *work*. I'm on Pipewire and can't figure out how to play audio. The JACK backend seems to work however. Can anyone confirm? @jcelerier could you update your PR to package ossia-score 3.2.2?

jcelerier commented:
Hmmm if you have pipewire running the jack backend likely just goes through pipewire too... So strange that it wouldn't work.
Is it through nixos, nix on another distro ?

And sure for the update, will do asap

jcelerier commented:
@minijackson one thing that comes to mind re. PW, is that ossia loads pipewire dynamically (through dlopen), would that influence anything with the way Nix works?
(note that it does the same for many other libraries, e.g. libasound, libjack, bluez etc)

jcelerier commented:
fixed the hash for 3.2.2 and also added a couple dependencies that can be fetched from nixpkgs instead of vendored (rubberband, snappy, and also avahi was missing avahi-compat-dnssd for full support)

jcelerier commented:
for pipewire it seems when I run the package that it cannot dlopen libpipewire ... is there some things that nix is doing that messes with the way dlopen resolves things ?

jcelerier commented:
e.g. I can see with strace that it's not looking in the folder where libpipewire is for dlopen: 

```

openat(AT_FDCWD, "/nix/store/971g0fka30djkcww2ddpx7q66dr86c1q-zlib-1.3.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/kq90rz7favab185f9lyabzxc6z0z95w8-libxkbcommon-1.7.0/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/a5vcvrkh1c2ng5kr584g3zw3991vnhks-libGL-1.7.0/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/vzhlcx5vzicjkbb9i37ryslaa6s3gb57-qtbase-6.7.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/3nmv7j8h47vlv13y34fnjl6j8rv323yz-qtshadertools-6.7.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/rqc91nln73pkmp4n6awx5pfhvv2kmla1-qtdeclarative-6.7.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/fj85yvx8zmakhf63xj0jkrnsih057514-ffmpeg-6.1.1-lib/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/pm7q9g2zxqk0n4xln21j0nbii1h2a7qj-fftw-double-3.3.10/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/0dbmgkb5pmsm178dfbyvgrdmirinib9p-fmt-10.2.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/bmx3p14nsy6y048aswkhraxpmdyc417v-gnutls-3.8.5/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/xj1l9pfz750f20z5gjpfamfvyaynn1zw-libsamplerate-0.2.2/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/yi30pd131302wivzrsywjr5sbydi4x1x-lilv-0.24.24/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/pfk8sbi90kwylvmvqy9y0z52kj8sg6ky-portaudio-190700_20210406/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/jcblgwdjzd3r4z46yy6njzb6mhr6a65y-qtserialport-6.7.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/cf5k2zfy4vsjg7bgwvmm5bx34vmi0jig-qtscxml-6.7.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/nbgb7ycijaajlmyvy6767y368zwjz1zf-qtsvg-6.7.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/psfcn5h41a4h3cj5kmq1fpfvmkp6fa2z-qtwebsockets-6.7.1/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/jx0491wq2p5rlxapjg1w4wdrwgmfiymp-re2-2024-05-01/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/qdb8j7p8slyxkqnm1wz4j1iyy68gvv52-SDL2-2.30.3/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/1hclbi4yhf7yx6hdfikq01rnxfwzvsjg-spdlog-1.13.0/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/h3kk9izdaxy9xaxkasvkvvjhp3mws1ra-suil-0.10.6/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/dbwp0scbb0rk78m636sb7cvycz8xzgyh-glibc-2.39-52/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/bn7pnigb0f8874m6riiw6dngsmdyic1g-gcc-13.3.0-lib/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/dbwp0scbb0rk78m636sb7cvycz8xzgyh-glibc-2.39-52/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/5k2fkrnnp9w7kjvf7s0ancb5rlnml100-xgcc-13.3.0-libgcc/lib/libpipewire-0.3.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
futex(0x7ad5a033e070, FUTEX_WAKE_PRIVATE, 2147483647) = 0
```

jcelerier commented:
I'm at loss, I've tried adding the following but strace still does not show dlopen going to look into the pipewire lib folder. pipewire is present in RUNPATH when I inspect the binary through readelf: 

```
(RUNPATH)            Library runpath: [/nix/store/f4ywrfccsr68b48dv5zwmsfrnxsxdpbh-libsndfile-1.2.2/lib:/nix/store/yi30pd131302wivzrsywjr5sbydi4x1x-lilv-0.24.24/lib:/nix/store/h3kk9izdaxy9xaxkasvkvvjhp3mws1ra-suil-0.10.6/lib:/nix/store/dbwp0scbb0rk78m636sb7cvycz8xzgyh-glibc-2.39-52/lib:/nix/store/crs0mhrx12zlwwcdijqw3ij9scl55hfr-pipewire-1.0.7/lib]
```

```nix
  runtimeDependencies = [
    alsa-lib
    avahi
    avahi-compat
    bluez
    fftw
    libjack2
    lilv
    pipewire
    udev
  ];
  
  postFixup = ''
    score_bin="$out/bin/ossia-score"
    oldrpath=$(patchelf --print-rpath "$score_bin")
    patchelf --set-rpath "$oldrpath:${lib.makeLibraryPath [ pipewire ]}" "$score_bin"
  '';

```

jcelerier commented:
okay, could make it work by passing -lpipewire-0.3 which does not make a lot of sense but if it makes nix happy we'll go with that

jcelerier commented:
@minijackson If you have some time to check again?

jcelerier commented:
up :)

Aleksanaa commented:
It looks like you added way more buildInputs than it actually needed. Could you please recheck?

jcelerier commented:
in particular a lot are ffmpeg & libsndfile codecs - the software can be used as a ffmpeg frontend so the more codecs available and the more varied file formats the software will be able to encode / decode / stream

Aleksanaa commented on pkgs/by-name/os/ossia-score/package.nix:
```diff
@@ -0,0 +1,162 @@
+    fftw
+    flac
+    fmt
+    git
```
But I guess we don't need this `git` or should put it in `nativeBuildInputs`?

It's used like this: https://github.com/ossia/libossia/blob/34f7737330f195fdb67214ea7f9a6cc038deaa0c/cmake/OssiaDeps.cmake#L70
```cmake
  execute_process(COMMAND ${GIT_EXECUTABLE} submodule sync --recursive
                  WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
                  COMMAND_ERROR_IS_FATAL ANY)
```

jcelerier replied:
Ah true,sorry I don't know exactly what the difference between these two keys is and just put it where it was needed to fix the build

jcelerier commented:
Updated package to use clangStdenv so that it supports aarch64

jcelerier commented:
arf, last change broke the build due to missing strstream when running with clang for some reason..

jcelerier commented:
Thanks <3
